TruePath/LocalPath.cs

```diff
@@ -42,7 +42,29 @@ namespace TruePath;
     /// <remarks>Note that currently this comparison is case-sensitive.</remarks>
     public bool Equals(LocalPath other)
     {
-        return Value == other.Value;
+        var comparer = PlatformDefaultPathComparer.Comparer;
+        return comparer.Compare(Value, other.Value) == 0;
+    }
+
+    /// <summary>
+    /// Determines whether the specified <see cref="LocalPath"/> is equal to the current <see cref="LocalPath"/> using the specified string comparer.
+    /// </summary>
+    /// <param name="other">The <see cref="LocalPath"/> to compare with the current <see cref="LocalPath"/>.</param>
+    /// <param name="comparer">The comparer to use for comparing the paths.</param>
+    /// <returns>
+    /// <see langword="true"/> if the specified <see cref="LocalPath"/> is equal to the current <see cref="LocalPath"/> using the specified string comparer; otherwise, <see langword="false"/>.
+    /// </returns>
+    /// <remarks>
+    /// If the comparer is null, this method returns <see langword="false"/>.
+    /// </remarks>
+    public bool Equals(LocalPath other, IComparer<string>? comparer)
+    {
+        if (comparer is null)
+        {
+            return false;
+        }
```

Comment:
Same comment as in `AbsolutePath`: let's consider making `comparer` non-nullable.